helix-core/src/main/java/org/apache/helix/common/cloud/CloudInformationFetcher.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.helix.common.cloud;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * An interface that provides cloud information fetching
+ * @param <T> The type of record to use
+ */
+public interface CloudInformationFetcher<T> {
```

Comment:
And be more accurate, this is  getVMInformation, right?

Reply:
In the interface, we don't want to be very specific, right? For Azure, the response actually has more than VM information. It also includes VMSS info, location info, network info etc. We can parse the response and get the fields we're interested in.

Reply:
This T type should extend CloudInstanceComputeInfo